### PR TITLE
lib9p now builds

### PR DESCRIPTION
**Pull request by keedon**
Signed-off-by: Keith Poole <keith.poole@gmail.com>

### CODE REVIEW

**Comment by keedon** [comment]
Now builds, but drawterm won't connect...

File: sys/include/9p.h [hunk 1]
```diff
@@ -120,13 +120,13 @@ void		dirread9p(Req*, Dirgen*, void*);
  * File trees.
  */
 struct File {
-	Ref;
-	Dir;
+	Ref ref;
+	Dir dir;
```
Comment (rminnich) [inline sys/include/9p.h]: Please change
ref -> Ref
dir -> Dir

File: sys/include/9p.h [hunk 2]
```diff
@@ -120,13 +120,13 @@ void		dirread9p(Req*, Dirgen*, void*);
...
 	File *parent;
 	void *aux;
 
 /* below is implementation-specific; don't use */
-	RWLock;
+	RWLock lock;
```
Comment (rminnich) [inline sys/include/9p.h]: RWLock

File: sys/src/lib9p/file.c [hunk 3]
```diff
@@ -72,12 +72,12 @@ freefile(File *f)
...
-	free(f->uid);
-	free(f->gid);
-	free(f->muid);
+	free(f->dir.name);
+	free(f->dir.uid);
+	free(f->dir.gid);
+	free(f->dir.muid);
 	qlock(&filelk);
-	assert(f->ref == 0);
+	assert(f->ref.ref == 0);
```
Comment (rminnich) [inline sys/src/lib9p/file.c]: this is fine, be sure not to change this ref b/c there is already ref (not Ref) in that struct.

File: sys/src/lib9p/file.c [hunk 4]
```diff
@@ -119,7 +119,7 @@ cleanfilelist(File *f)
 void
 closefile(File *f)
 {
-	if(decref(f) == 0){
+	if(decref(&f->ref) == 0){
```
Comment (rminnich) [inline sys/src/lib9p/file.c]: Ref

File: sys/src/lib9p/file.c [hunk 5]
```diff
@@ -149,20 +149,20 @@ removefile(File *f)
 		return -1;
 	}
 
-	wlock(f);
-	wlock(fp);
+	wlock(&f->lock);
```
Comment (rminnich) [inline sys/src/lib9p/file.c]: Lock etc. etc

File: sys/src/lib9p/file.c [hunk 6]
```diff
@@ -220,36 +220,36 @@ createfile(File *fp, char *name, char *uid, uint32_t perm, void *aux)
 	*l = fl;
 
 	f = allocfile();
-	f->name = estrdup9p(name);
-	f->uid = estrdup9p(uid ? uid : fp->uid);
-	f->gid = estrdup9p(fp->gid);
-	f->muid = estrdup9p(uid ? uid : "unknown");
+	f->dir.name = estrdup9p(name);
```
Comment (rminnich) [inline sys/src/lib9p/file.c]: Dir everywhere of course.

**Comment by rminnich** [comment]
-1
Please make the changes I mention, this is just about perfect!

**Pull request closed without merging**